### PR TITLE
feat(headless): expand and collapse methods added to generated answer controller

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-5505](https://coveord.atlassian.net/browse/SFINT-5505)

In this PR: 

- coveo.analytics.js version updated. (coming soon).
- Added the `expand` and `collapse` methods to the Headless Generated Answer controller. These new methods will be useful for implementing the feature of expanding and collapsing a generated answer in Atomic and in Quantic.
- Added the analytics actions linked to these new two actions.
- Added uni tests to cover the new logic.

[SFINT-5505]: https://coveord.atlassian.net/browse/SFINT-5505?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/controllers/core/generated-answer/headless-core-generated-answer.test.ts [hunk 1]
```diff
@@ -227,6 +231,52 @@ describe('generated answer', () => {
...
   });
 
+  describe('#expand', () => {
+    describe('when already expanded', () => {
+      it('should not make any changes', () => {
+        engine = buildEngineWithGeneratedAnswer({expanded: true});
+        initGeneratedAnswer();
+        generatedAnswer.expand();
+
+        expect(expandGeneratedAnswer).not.toHaveBeenCalled();
```
Comment (mmitiche) [inline packages/headless/src/controllers/core/generated-answer/headless-core-generated-answer.test.ts:241]:
```suggestion
        expect(expandGeneratedAnswer).not.toHaveBeenCalled();
        expect(logGeneratedAnswerExpand).not.toHaveBeenCalled();
```

File: packages/headless/src/controllers/core/generated-answer/headless-core-generated-answer.test.ts [hunk 2]
```diff
@@ -227,6 +231,52 @@ describe('generated answer', () => {
...
+  });
+
+  describe('#collapse', () => {
+    describe('when is not expanded', () => {
+      it('should not make any changes', () => {
+        engine = buildEngineWithGeneratedAnswer({expanded: false});
+        initGeneratedAnswer();
+        generatedAnswer.collapse();
+
+        expect(collapseGeneratedAnswer).not.toHaveBeenCalled();
```
Comment (mmitiche) [inline packages/headless/src/controllers/core/generated-answer/headless-core-generated-answer.test.ts:264]:
```suggestion
        expect(collapseGeneratedAnswer).not.toHaveBeenCalled();
        expect(logGeneratedAnswerCollapse).not.toHaveBeenCalled();
```

File: packages/headless/src/controllers/core/generated-answer/headless-core-generated-answer.test.ts [hunk 3]
```diff
@@ -227,6 +231,54 @@ describe('generated answer', () => {
...
+        generatedAnswer.expand();
+
+        expect(expandGeneratedAnswer).toHaveBeenCalled();
+        expect(logGeneratedAnswerExpand).toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe('#collapse', () => {
+    describe('when is not expanded', () => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/core/generated-answer/headless-core-generated-answer.test.ts:259]:
```suggestion
    describe('when not expanded', () => {
```

File: packages/headless/src/features/analytics/search-action-cause.ts [hunk 4]
```diff
@@ -340,6 +340,14 @@ export enum SearchPageEvents {
    * Identifies the custom event that gets logged when a user activates the RGA feature.
    */
   generatedAnswerShowAnswers = 'generatedAnswerShowAnswers',
+  /**
+   * Identifies the custom event that gets logged when a user expand a generated answer.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/features/analytics/search-action-cause.ts:344]:
```suggestion
   * Identifies the custom event that gets logged when a user expands a generated answer.
```

File: packages/headless/src/features/analytics/search-action-cause.ts [hunk 5]
```diff
@@ -340,6 +340,14 @@ export enum SearchPageEvents {
...
+   */
+  generatedAnswerExpand = 'generatedAnswerExpand',
+  /**
+   * Identifies the custom event that gets logged when a user collapse a generated answer.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/features/analytics/search-action-cause.ts:348]:
```suggestion
   * Identifies the custom event that gets logged when a user collapses a generated answer.
```

File: packages/headless/src/features/generated-answer/generated-answer-state.ts [hunk 6]
```diff
@@ -59,6 +59,10 @@ export interface GeneratedAnswerState {
    * Determines if the answer is generated.
    */
   isAnswerGenerated: boolean;
+  /**
+   * Determines if the answer is expanded.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/features/generated-answer/generated-answer-state.ts:63]:
```suggestion
   * Whether the answer is expanded.
```

**Review by fbeaudoincoveo — APPROVED** [review]

**Review by olamothe — APPROVED** [review]

**Review by SimonMilord — APPROVED** [review]

**Review by erocheleau — APPROVED** [review]
👍

File: packages/headless/src/controllers/core/generated-answer/headless-core-generated-answer.test.ts [hunk 7]
```diff
@@ -227,6 +231,52 @@ describe('generated answer', () => {
...
+      it('should not make any changes', () => {
+        engine = buildEngineWithGeneratedAnswer({expanded: false});
+        initGeneratedAnswer();
+        generatedAnswer.collapse();
+
+        expect(collapseGeneratedAnswer).not.toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/headless/src/controllers/core/generated-answer/headless-core-generated-answer.test.ts:265]: indeed!

**Pull request merged**